packages/nextjs/src/create-server-side-client.ts

```diff
@@ -0,0 +1,44 @@
+import Cookies from 'cookies'
+import { GetServerSidePropsContext } from 'next'
+import { StateFrom } from 'xstate'
+import { waitFor } from 'xstate/lib/waitFor'
+
+import { AuthMachine } from '@nhost/core'
+import { NhostClient } from '@nhost/nhost-js'
+
+/**
+ * Creates an Nhost client that runs on the server side.
+ * It will try to get the refesh token in cookies, or from the request URL
+ * If a refresh token is found, it uses it to get an up to date access token (JWT) and a user session
+ * This method resolves when the authentication status is known eventually
+ * @param backendUrl
+ * @param context
+ * @returns
+ */
+export const createServerSideClient = async (
+  backendUrl: string,
+  context: GetServerSidePropsContext
+) => {
+  const cookies = Cookies(context.req, context.res)
+  const nhost = new NhostClient({
```

Comment:
Would be great if people could provide some options to this internal `NhostClient`. Even if it's just a limited set of options, we should let them tweak with things like `refreshIntervalTime`, etc. We shouldn't let people override crucial options though, such as `clientStorageGetter`.

For example:
```ts
const ssrClient = createServerSideClient('https://<my-url>.nhost.run', ctx, {
  refreshIntervalTime: 5000
})
```

What do you think @plmercereau ? Is it something that we wish to use only internally?

Reply:
This method is meant to be called once when rendering on the server side, so we don't want activate the auto refresher - `refreshIntervalTime` being related to this. Plus we overload the refresh toker getter/setter, so there's no other option left for now.